### PR TITLE
Displayed only board members in autocomplete

### LINKED ISSUE #2886: Bug: Auto-complete list displays users from all boards viewed
## Steps to reproduce the behavior

1. Go to any board with several members (Board A)
2. Open a card and go to the Comments section
3. @mention to load auto-complete list
4. Go to a private board with a different set of board members (Board B)
5. Open a card and go to the Comments section
6. @mention to load auto-complete list
7. See error: Auto-complete lists users from Board A and Board B, even though some users may not even be on the team for Board B
8. Refreshing the browser seems to reset this behavior for the current board

## Expected behavior

Public boards should list all team members and private boards should only list board members. See [spec](https://docs.google.com/document/d/1HOnoB87WZojqplpB6XGbtKsLuqjG8h-8CbJpmWsRfM4/edit#bookmark=kix.p1mww7cnoiz4) for details.

## Screenshots (optional)

https://user-images.githubusercontent.com/93531870/164530895-f81948c5-9acf-44e2-b780-2ca227d4cc93.mov

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: v0.16 Bugbash-2
 - Browser and OS: Chrome on Mac

### PR DESCRIPTION
**Pull request by harshilsharma63**
Fixes #2886

The issue was that we were adding new board members to the store who weren't part of the previous board, but not removing old users, who aren't part of the new board.

### CODE REVIEW

**Comment by harshilsharma63** [comment]
@sbishel can you review this, please?

File: webapp/src/components/cardDialog.tsx [hunk 1]
```diff
@@ -179,6 +179,8 @@ const CardDialog = (props: Props): JSX.Element => {
     const isFollowingCard = Boolean(followingCards.find((following) => following.blockId === props.cardId))
     const toolbar = followActionButton(isFollowingCard)
 
+    console.log(useAppSelector(getBoardUsers))
```
Comment (sbishel) [inline webapp/src/components/cardDialog.tsx:182]: Probably want to remove.

File: webapp/src/store/boards.ts [hunk 2]
```diff
@@ -25,23 +25,23 @@ type BoardsState = {
 export const fetchBoardMembers = createAsyncThunk(
     'boardMembers/fetch',
     async ({teamId, boardId}: {teamId: string, boardId: string}, thunkAPI: any) => {
+        const allTeamUsers = await octoClient.getTeamUsers()
+        console.log(allTeamUsers)
```
Comment (sbishel) [inline webapp/src/store/boards.ts:29]: Doesn't look like `allTeamUsers` is used, except for log message. These two lines can be removed.

**Review by sbishel — CHANGES_REQUESTED** [review]

File: webapp/src/components/cardDialog.tsx (hunk 1 above)
Comment (harshilsharma63) [reply webapp/src/components/cardDialog.tsx]: Done.

File: webapp/src/store/boards.ts (hunk 2 above)
Comment (harshilsharma63) [reply webapp/src/store/boards.ts]: Done.

**Comment by harshilsharma63** [comment]
@sbishel I've removed the debug logs.

**Review by sbishel — APPROVED** [review]

**Pull request merged**